### ISSUE TITLE
IDOR in "/userinfo.php?uid=1" endpoint in version 1.3.11

### ISSUE THREAD

**Issue by Akshay-Kawlay** [ImpressCMS/impresscms#1081]
## Summary:
Regular user can access another user's private data like name, email, occupation, rank, last login datetime..etc by replacing the uid value to the victim's uid

## ImpressCMS branch :
1.3.11
## Browsers Verified In:

Chrome Version 95.0.4638.69

## Steps To Reproduce:
Visit endpoint /userinfo.php?uid=1 and replace the uid value to victim's uid

## Suggestions to mitigate or resolve the issue:
Enforce proper access controls to the endpoint

## Impact
Information Disclosure of private data

**Comment by fiammybe** [comment]
Hi, ImpressCMS 1.3 and 1.4 allow access to this information for logged in users. That is the way the system is conceived, and is not a vulnerability.
As a matter of fact, 'real name' is not a required field, and the user must explicitly check the box to show his mail address to the outside world. So email is NOT visible by default, not even to logged-in users.

**Comment by Akshay-Kawlay** [comment]
It is still a vulnerability even if no private data is disclosed by default. The fact that changing uid in url to someone else's gets a valid response from the server shows that access control checks for that resource are absent. Ideally, doing this should be forbidden.
Below is the screenshot of the response when userB makes an IDOR to userA's userinfo -

![Capture](https://user-images.githubusercontent.com/24737270/141367965-f0705756-10ca-47cf-bc9e-60b5e981489c.PNG)

**Comment by MekDrop** [comment]
We talked about this internally. And yeah by today's standards this way how this works is not very good. But when this was written there weren't any issues with that. And this design desition was implemented in such way that now is not easy to change that. So, we have decided to not change the way how it works for *1.x* releases and do that only for 2.0. 

Of course, if somebody would create a pull request for this issue to *1.x* branch we probably would try to add it to next our release.
